controllers/instancechartrepository_controller.go

```diff
@@ -108,18 +111,38 @@ func (r *InstanceChartRepositoryReconciler) SetupWithManager(mgr ctrl.Manager) e
 		return err
 	}
 
-	// Watch for changes to secondary resource Pods and requeue the owner InstanceChartRepo
-	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &registriesv1alpha1.InstanceChartRepository{},
-	})
+	// Watch for changes to the secondary resource corev1.Secrets
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{})
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// reconcileInstanceChartRepositorySecret fetches the secret specified in an
+// InstanceChartRepository's spec and sets an OwnerReference to the owned Object.
+// Returns nil when the OwnerReference has been successfully set, or when no secret is specified.
+func (r *InstanceChartRepositoryReconciler) reconcileInstanceChartRepositorySecret(ctx context.Context, i *registriesv1alpha1.InstanceChartRepository) error {
+	if i.Spec.SecretRef != nil {
+		secret, err := r.getSecret(ctx, i)
+		if err != nil {
+			return err
+		}
+
+		if secret.OwnerReferences == nil || len(secret.OwnerReferences) == 0 {
```

Comment:
```suggestion
		if len(secret.OwnerReferences) == 0 {
```
len will also return `0` for `nil` slices